CheckRightsService_Database

nmamro opened the pull request.

nmamro committed:
Some initial stuff

nmamro committed:
Updated ports

nmamro committed:
Some more code describing db

nmamro committed:
Added links

MihailKulikov commented on DigitalOffice/Services/CheckRightsService/CheckRightsService.csproj:
```diff
@@ -0,0 +1,17 @@
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.EntityFrameworkCore" Version="5.0.0-preview.6.20312.4" />
+    <PackageReference Include="Microsoft.EntityFrameworkCore.SqlServer" Version="5.0.0-preview.6.20312.4" />
```
Is it a good practise to use preview versions of the packages?

mekk1t replied:
I think not. 5.0 update will be released only in September and we need to work on the stable version.
Also, how is this even possible? NuGet prevents this kind of versioning and allows to install only stable versions.
![image](https://user-images.githubusercontent.com/61093487/87470543-58e29780-c636-11ea-958e-bfb526f40c88.png)

nmamro replied:
> I think not. 5.0 update will be released only in September and we need to work on the stable version.
> 
> Also, how is this even possible? NuGet prevents this kind of versioning and allows to install only stable versions.
> 
> ![image](https://user-images.githubusercontent.com/61093487/87470543-58e29780-c636-11ea-958e-bfb526f40c88.png)
> 
> 

Yikes. No problem changing it, but usually previews which are already accessible r ok, at least haven't heard of possible "disasters" caused by using them:) However, might be mistaken:)

denis-kriv commented on DigitalOffice/Services/CheckRightsService/Database/Entities/RightChangeRecord.cs:
```diff
@@ -0,0 +1,21 @@
+        [Required]
+        public Guid RightId { get; set; }
+        [Required]
+        public Guid ChangedById { get; set; }
```
Maybe we need `ChangedByUserId` here?

nmamro replied:
> Maybe we need `ChangedByUserId` here?

Why "user" part would be necessary?

KostinP replied:
Since the name of the field does not match the name of the field you are referring to.

mekk1t replied:
> > Maybe we need `ChangedByUserId` here?
> 
> Why "user" part would be necessary?

We decided to make it clear which Ids we are referring to.
In this case, we can't tell exactly at what Id we're looking: is it a FileId? A ProjectId? So we mark it as UserId.

denis-kriv reviewed (commented):
Resolve conflicts please

nmamro committed:
Wow stable packages' versions

nmamro committed:
Merge branch 'develop' into feature/CheckRightsService_Base

KostinP commented on DigitalOffice/Services/CheckRightsService/Database/Entities/RightChangeRecordProjectLink.cs:
```diff
@@ -0,0 +1,15 @@
+
+namespace CheckRightsService.Database.Entities
+{
+    public class RightChangeRecordProjectLink
```
Why do we need a separate class?

KostinP replied:
Thanks for the dialogue in Slack, it became clear what you meant. I have a new question about Type. As I understand it, there may be the right to edit "X" (for example, the description for the project). And we will have to do RightToEditDescriptionToProject, maybe we should put the Type in a linking table?

nmamro replied:
> Thanks for the dialogue in Slack, it became clear what you meant. I have a new question about Type. As I understand it, there may be the right to edit "X" (for example, the description for the project). And we will have to do RightToEditDescriptionToProject, maybe we should put the Type in a linking table?

Check out latest changes, I guess there're right types table u r talking about. If not, comment again :)

KostinP replied:
I saw the following entities:
- DbRightType (Id, RightType Type, ICollection<RightTypeLink> Rights, ICollection<RightChangeRecordTypeLink> RightChangeRecords)
- Right(Id, UserId, ICollection<RightTypeLink> Types, ICollection<RightProjectLink> PermissionsIds)
- RightChangeRecord(Id, RightId, ChangedByUserId, DateTime Time, ICollection<RightChangeRecordTypeLink> Types, ICollection<RightRecordProjectLink> ChangedPermissionsIds)
- RightChangeRecordTypeLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, RightTypeId, DbRightType RightType)
- RightProjectLink(RightId, Right Right, ProjectId)
- RightRecordProjectLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, ProjectId)
- RightTypeLink(RightId, Right Right, RightTypeId, DbRightType RightType)
-  enum RightType 

And they got me confused. Maybe we should add a folder for the linking class. Maybe we should change the naming convention. I propose to discuss this at the general call.

KostinP replied:
And here the linking class convention is broken (Entity1Entity2 is a class linking the Entity1 and the Entity2)
Maybe we should change the naming convention this way: Entity1_Entity2
P.S. Perhaps I confused something and this convention was only about tables

zzzloba replied:
> I saw the following entities:
> 
> * DbRightType (Id, RightType Type, ICollection Rights, ICollection RightChangeRecords)
> * Right(Id, UserId, ICollection Types, ICollection PermissionsIds)
> * RightChangeRecord(Id, RightId, ChangedByUserId, DateTime Time, ICollection Types, ICollection ChangedPermissionsIds)
> * RightChangeRecordTypeLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, RightTypeId, DbRightType RightType)
> * RightProjectLink(RightId, Right Right, ProjectId)
> * RightRecordProjectLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, ProjectId)
> * RightTypeLink(RightId, Right Right, RightTypeId, DbRightType RightType)
> * enum RightType
> 
> And they got me confused. Maybe we should add a folder for the linking tables. Maybe we should change the naming convention. I propose to discuss this at the general call.

Ok np with discussion :)

nmamro replied:
> I saw the following entities:
> 
> * DbRightType (Id, RightType Type, ICollection Rights, ICollection RightChangeRecords)
> * Right(Id, UserId, ICollection Types, ICollection PermissionsIds)
> * RightChangeRecord(Id, RightId, ChangedByUserId, DateTime Time, ICollection Types, ICollection ChangedPermissionsIds)
> * RightChangeRecordTypeLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, RightTypeId, DbRightType RightType)
> * RightProjectLink(RightId, Right Right, ProjectId)
> * RightRecordProjectLink(RightChangeRecordId, RightChangeRecord RightChangeRecord, ProjectId)
> * RightTypeLink(RightId, Right Right, RightTypeId, DbRightType RightType)
> * enum RightType
> 
> And they got me confused. Maybe we should add a folder for the linking class. Maybe we should change the naming convention. I propose to discuss this at the general call.

Discussion is a good thing

nmamro replied:
> And here the linking class convention is broken (Entity1Entity2 is a class linking the Entity1 and the Entity2)
> Maybe we should change the naming convention this way: Entity1_Entity2

Mb I messed up with following naming convension, not sure :)

DarkFelas commented on DigitalOffice/Services/CheckRightsService/Database/CheckRightsServiceDbContext.cs:
```diff
@@ -0,0 +1,40 @@
+
+        public CheckRightsServiceDbContext(DbContextOptions<CheckRightsServiceDbContext> options) : base(options) { }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
```
Please use [separate configuration classes](https://www.learnentityframeworkcore.com/configuration/fluent-api#separate-configuration-classes). Put config class in the same file where model is.

nmamro replied:
> Please use [separate configuration classes](https://www.learnentityframeworkcore.com/configuration/fluent-api#separate-configuration-classes). Put config class in the same file where model is.

Added config classes

DarkFelas replied:
use ApplyConfigurationsFromAssembly(Assembly.GetExecutingAssembly());

DarkFelas commented on DigitalOffice/Services/CheckRightsService/Startup.cs:
```diff
@@ -0,0 +1,58 @@
+
+        private void UpdateDatabase(IApplicationBuilder app)
+        {
+            using (var scope = app.ApplicationServices
```
use new version of using please

nmamro replied:
> use new version of using please

Refactored for new version

nmamro committed:
Added right types linking

Added configurations classes for entities
Added linking between rights and types
Rewrote linking for projects

nmamro committed:
Moved configurations to models' files

nmamro committed:
Merge branch 'develop' into feature/CheckRightsService_Base

nmamro committed:
Moved projects to same sln directory

nmamro committed:
Removed extra table corresponding right types

nmamro committed:
Switched to applying all configs from the assembly

nmamro committed:
Merge branch 'develop' into feature/CheckRightsService_Base

nmamro committed:
Update DigitalOffice.sln

MihailKulikov commented on DigitalOffice/Services/CheckRightsService/Database/Entities/RightTypeLink.cs:
```diff
@@ -0,0 +1,31 @@
+                .HasForeignKey(link => link.RightId);
+
+            builder
+            .Property(e => e.RightType)
```
Indentation issues :)

nmamro replied:
> Indentation issues :)

Thanks, reformatted :)

nmamro committed:
Fixed indentations

nmamro committed:
Merge branch 'develop' into feature/CheckRightsService_Base

nmamro committed:
Updated solution structure

mekk1t reviewed (approved):
All classes are correctly designed as are Startup.cs, DbContext and separate configuration classes.
Need to refactor the names though and to discuss the whole service, 'cause I don't like how we implemented it.
I have the doubts because of the yesterday's call. The implementation there seemed much easier, cleaner and better.

mekk1t committed:
Merge branch 'develop' into feature/CheckRightsService_Base

mekk1t committed:
Update DigitalOffice.sln

Fixed indentation in .sln file by adding a few spaces.
